### ISSUE TITLE
a11y: Failing color contrast in prism themes

### ISSUE THREAD

**Issue by ehowey** [FormidableLabs/prism-react-renderer#62]
Hey All,

This is related to: https://github.com/system-ui/theme-ui/issues/564

I am using [Theme-UI](https://theme-ui.com/) in some Gatsby projects. I installed Cypress for testing and [cypress axe](https://www.npmjs.com/package/cypress-axe) for accessibility testing. Turns out that some of the color contrasts on the prism themes are failing a11y checks. In most cases the color contrasts were meeting AA Large but because of the font size need to meet AA. I dug into a bit, you can see that in my above issue, originally made a PR for Theme-UI however didn't realize they are just auto-generating from your presets.

I would be happy to take it on, but don't know much about this project. Or how you would want a fix to look.

**Comment by kitten** [comment]
Hiya 👋

Most of these themes are either converted manually from Prism themes or they’re auto-generated from VSCode themes.

It’s probably not a good idea to just alter them since people are depending on them being consistent, but you could tweak some and make accessible variants maybe?

The readme could then maybe list out all themes in a table and list their AA/AAA contrast compliance. 🤔

**Comment by knitcodemonkey** [comment]
Though I agree that themes should be accessible by default, there are those that put in the work to make compatible accessible themes. As `react-prism-renderer` is a react wrapper around  `PrismJS`, you can pass in new theme files, such as [a11y-syntax-highlighting](https://openbase.io/js/a11y-syntax-highlighting) to get the accessible theming you, and many others, are looking for.

**Comment by ehowey** [comment]
Thanks for this.  I may just close this issue out for now.  I don't have the time to work on it unfortunately.  Thank you for that link to the theme file though!  I will look into implementing that for my site!
